packages/react-devtools-core/src/deviceStorage.js

```diff
@@ -0,0 +1,106 @@
+/**
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ *
+ * @flow
+ */
+
+import {
+  type ConsolePatchSettings,
+  writeConsolePatchSettingsToWindow,
+} from 'react-devtools-shared/src/backend/console';
+import {
+  LOCAL_STORAGE_SHOULD_APPEND_COMPONENT_STACK_KEY,
+  LOCAL_STORAGE_SHOULD_BREAK_ON_CONSOLE_ERRORS,
+  LOCAL_STORAGE_SHOW_INLINE_WARNINGS_AND_ERRORS_KEY,
+  LOCAL_STORAGE_HIDE_CONSOLE_LOGS_IN_STRICT_MODE,
+  LOCAL_STORAGE_BROWSER_THEME,
+} from 'react-devtools-shared/src/constants';
+import type {BrowserTheme} from 'react-devtools-shared/src/devtools/views/DevTools';
+
+export type DeviceStorageMethods = {|
+  setValueOnDevice: (key: string, value: string) => void,
+  getValueOnDevice: (key: string) => ?string,
+|};
+
+export function initializeFromDeviceStorage(methods: DeviceStorageMethods) {
+  writeDeviceStorageValuesToWindow(methods);
+}
+
+function writeDeviceStorageValuesToWindow(methods: DeviceStorageMethods) {
+  const settings: ConsolePatchSettings = {
+    appendComponentStack:
+      parseBool(
+        methods.getValueOnDevice(
+          LOCAL_STORAGE_SHOULD_APPEND_COMPONENT_STACK_KEY,
+        ),
+      ) ?? true,
+    breakOnConsoleErrors:
+      parseBool(
+        methods.getValueOnDevice(LOCAL_STORAGE_SHOULD_BREAK_ON_CONSOLE_ERRORS),
+      ) ?? false,
+    showInlineWarningsAndErrors:
+      parseBool(
+        methods.getValueOnDevice(
+          LOCAL_STORAGE_SHOW_INLINE_WARNINGS_AND_ERRORS_KEY,
+        ),
+      ) ?? true,
+    hideConsoleLogsInStrictMode:
+      parseBool(
+        methods.getValueOnDevice(
+          LOCAL_STORAGE_HIDE_CONSOLE_LOGS_IN_STRICT_MODE,
+        ),
+      ) ?? false,
+    browserTheme:
+      parseBrowserTheme(
+        methods.getValueOnDevice(LOCAL_STORAGE_BROWSER_THEME),
+      ) ?? 'dark',
+  };
+  writeConsolePatchSettingsToWindow(settings);
+}
+
+function parseBool(s: ?string): ?boolean {
+  if (s === 'true') {
+    return true;
+  }
+  if (s === 'false') {
+    return false;
+  }
+}
+
+function parseBrowserTheme(s: ?string): ?BrowserTheme {
+  if (s === 'light') {
+    return 'light';
+  }
+  if (s === 'dark') {
+    return 'dark';
+  }
+}
```

Comment:
IIRC we are casting/parsing browser theme and bools for both local storage and device storage. Can we maybe pull these out into utils?

Reply:
good catch